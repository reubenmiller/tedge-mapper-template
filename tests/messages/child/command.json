{
    "delivery": {
        "log": [],
        "time": "2023-05-26T20:10:54.494Z",
        "status": "PENDING"
    },
    "agentId": "822039111",
    "creationTime": "2023-05-26T20:10:54.446Z",
    "deviceId": "372037885",
    "id": "2042018",
    "status": "PENDING",
    "c8y_Command": {
        "text": "ls -l"
    },
    "description": "Execute shell command",
    "externalSource": {
        "externalId": "sim_tedge01:device:child02",
        "type": "c8y_Serial"
    }
}