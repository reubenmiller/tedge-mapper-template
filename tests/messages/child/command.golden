Route: c8y-devicecontrol-notifications (c8y/devicecontrol/notifications)

Input Message
  topic:    c8y/devicecontrol/notifications

Output Message (mqtt)
  topic:    c8y/devicecontrol/notifications/sim_tedge01:device:child02/c8y_Command

    {
      "_ctx": {
        "agentID": "822039111",
        "deviceID": "372037885",
        "id": "2042018",
        "localSerial": "sim_tedge01:device:child02",
        "lvl": 1,
        "opType": "c8y_Command",
        "operationID": "2042018",
        "parent": "sim_tedge01",
        "serial": "sim_tedge01:device:child02"
      },
      "payload": {
        "agentId": "822039111",
        "c8y_Command": {
          "text": "ls -l"
        },
        "creationTime": "2023-05-26T20:10:54.446Z",
        "delivery": {
          "log": [],
          "status": "PENDING",
          "time": "2023-05-26T20:10:54.494Z"
        },
        "description": "Execute shell command",
        "deviceId": "372037885",
        "externalSource": {
          "externalId": "sim_tedge01:device:child02",
          "type": "c8y_Serial"
        },
        "id": "2042018",
        "status": "PENDING"
      }
    }

Route: shell-operation (c8y/devicecontrol/notifications/+/c8y_Command)

Input Message
  topic:    c8y/devicecontrol/notifications/sim_tedge01:device:child02/c8y_Command

Output Message (mqtt)
  topic:    te/device/child02///cmd/execute_shell/c8y-2042018

    {
      "command": "ls -l"
    }